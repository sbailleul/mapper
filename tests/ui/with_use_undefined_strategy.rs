use mapper_impl::Mapper;

fn with_test(val: &String)->String{
    val
}

#[derive(Mapper)]
#[to(Person,)]
pub struct User{
    #[to(Person, with=with_test)]
    pub name: String
}

pub struct Person{
    pub name: String
}

fn main(){}