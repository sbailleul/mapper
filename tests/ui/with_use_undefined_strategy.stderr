error: There is no destination and strategy matching with strategy (mapper) and destination (Person)
  --> tests/ui/with_use_undefined_strategy.rs:10:23
   |
10 |     #[to(Person, with=with_test)]
   |                       ^^^^^^^^^

error[E0308]: mismatched types
 --> tests/ui/with_use_undefined_strategy.rs:4:5
  |
3 | fn with_test(val: &String)->String{
  |                             ------ expected `String` because of return type
4 |     val
  |     ^^^- help: try using a conversion method: `.to_string()`
  |     |
  |     expected struct `String`, found `&String`
